packages/frontend/src/components/Explorer/ExploreTree/index.tsx

```diff
@@ -71,12 +72,30 @@ const ExploreTree: FC<ExploreTreeProps> = ({
     }, [explore, searchHasResults, isSearching]);
 
     const missingCustomMetrics = useMemo(() => {
-        const allTables = Object.keys(explore.tables);
-        return additionalMetrics.filter(
-            (metric) => !allTables.includes(metric.table),
-        );
+        return additionalMetrics.filter((metric) => {
+            const table = explore.tables[metric.table];
+            return (
+                !table ||
+                (metric.baseDimensionName &&
+                    !table.dimensions[metric.baseDimensionName])
+            );
+        });
     }, [explore, additionalMetrics]);
 
+    const missingCustomDimensions = useMemo(() => {
+        return customDimensions?.filter((customDimension) => {
+            const table = explore.tables[customDimension.table];
+
+            return (
+                !table ||
+                (isCustomBinDimension(customDimension) &&
```

Comment:
do we also need to check if `isCustomSqlDimension` ?

Reply:
good point, I had a look at that and `customSqlDimension` but since it doesn't have a base dimension and it's only sql I couldn't do it like this, probably can only do it after compiling?

Reply:
I believe I was able to do this in the latest [commit](https://github.com/lightdash/lightdash/pull/10070/commits/f8dd428263461d83a9f87cf9f99776c92496bbc1): 


https://github.com/lightdash/lightdash/assets/22939015/dfc2232b-fb50-4289-baca-4f5ebed702a7